### PR TITLE
V1.0.0

### PR DESCRIPTION
**Pull request by juliancoleman**
literally the fattest update, so yeah...

### CODE REVIEW

File: vigor/forms.py [hunk 1]
```diff
@@ -1,20 +1,57 @@
...
+            'time_zone',
+            'date_format'
+        )
+
+
+class ProfileForm(forms.ModelForm):
+    class Meta:
+        model = UserPrefs
+        fields = ('gender', 'birth_year',)
+# from django.contrib.auth.forms import UserCreationForm
```
Comment (ebramanti) [inline vigor/forms.py]: @juliancoleman If there's no reason to keep this commented out stuff around, let's get rid of it! 😄

File: vigor/static/css/style.css [hunk 2]
```diff
@@ -0,0 +1,73 @@
+@import "../bower_components/material-typography/css/typography.css";
```
Comment (ebramanti) [inline vigor/static/css/style.css:1]: @juliancoleman I added this, maybe its breaking typography?

**Comment by ebramanti** [comment]
@juliancoleman I've given it a once-over... most of this is out of my domain of knowledge! 😄 

Overall, impressed with Django. Impressed with your progress. I wanna work with you to get this deployed on Heroku with continuous integration.

@birkholz Would you mind helping @juliancoleman out with some review on this?

File: vigor/ajax.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        if obj is None:
+            obj = {}
+        if json_opts is None:
+            json_opts = {}
+        content = json.dumps(obj, **json_opts)
+        super(JSONResponse, self).__init__(content, mimetype, *args, **kwargs)
+
+def sign_out(req):
+    logout(req)
+    # next = req.GET.get('next', None)
```
Comment (ebramanti) [inline vigor/ajax.py]: @juliancoleman Blow out this comment too

File: vigor/ajax.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    if settings.REGISTER:
+        if req.method == "POST":
+            username = req.POST['username']
+            password = req.POST['password']
+            email = req.POST['email']
+
+            user = User.objects.create_user(username, email, password)
+            UserPrefs.objects.create(user = user)
+            user2 = authenticate(username = username, password = password)
+            login(req, user2)
```
Comment (ebramanti) [inline vigor/ajax.py:44]: @juliancoleman 

```
login(req, authenticate(username = username, password = password))
```

Improvement? `user2` is a pretty nebulous variable name

File: vigor/forms.py [hunk 5]
```diff
@@ -1,20 +1,41 @@
...
-        if email and User.objects.filter(email=email).count():
-            raise forms.ValidationError('This email is already bound to an existing profile')
-        return email
 
+class ProfileForm(forms.ModelForm):
     class Meta:
-        model = User
-        fields = ('username', 'email', 'first_name', 'last_name')
+        model = UserPrefs
+        fields = ('gender', 'birth_year',)
```
Comment (ebramanti) [inline vigor/forms.py]: @juliancoleman Need that comma?

File: vigor/models.py [hunk 6]
```diff
@@ -1,11 +1,129 @@
...
+        elif self.determination == 'Maintain Weight':
+            if self.gender == 'M':
+                return 10 * self.weight + 6.25 * (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() + 5
+            elif self.gender == 'F':
+                return 10 * self.weight + 6.25 (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() - 161
+        elif self.determination == 'Lose Weight':
+            if self.gender == 'M':
+                return 10 * self.weight + 6.25 * (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() + 5 - 500
+            elif self.gender == 'F':
+                return 10 * self.weight + 6.25 (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() - 161 - 500
```
Comment (ebramanti) [inline vigor/models.py]: @juliancoleman This method is very complex. I'm not sure how Django handles breaking this out, but it would be nice to make this some kind of service.

cc @birkholz

Comment (ebramanti) [inline vigor/models.py]: @juliancoleman Should be fine for now 😄

File: vigor/urls.py [hunk 7]
```diff
@@ -2,14 +2,11 @@
...
-    url(r'^logout/$', views.logout, name='logout'),
-    url(r'^dashboard/$', views.dashboard, name='dashboard'),
-    url(r'^profile/$', views.profile, name='profile'),
-    url(r'^add_item/$', views.add_item, name='add_item'),
-    url(r'^users/$', views.users, name='users')
+    url(r'^$', 'vigor.views.index', name='index'),
+    url(r'^sign_out/$', 'vigor.ajax.sign_out', name='sign_out'),
+    url(r'^signin/$', 'vigor.views.login_view', name='login'),
+    url(r'^register/$', 'vigor.ajax.register', name='register'),
+    url(r'^preferencess/$', 'vigor.views.preferences', name='preferences'),
```
Comment (ebramanti) [inline vigor/urls.py]: @juliancoleman Whoops 😄

File: vigor/static/css/style.css (hunk 2 above)
Comment (juliancoleman) [inline vigor/static/css/style.css:1]: You know what? I actually have material called in the actual `style.css` file. Maybe that's what's happening, too

Comment (juliancoleman) [inline vigor/static/css/style.css:1]: wait. you are importing it in the CSS... Maybe Bower hasn't updated the versions yet, and it's still on 1.0.7 before all those typography updates.

File: vigor/ajax.py (hunk 4 above)
Comment (juliancoleman) [inline vigor/ajax.py:44]: I think the reason why I went with this is because I'm writing my own authentication since I have to extend the `User` model. I have to create a custom user. But this is also how the Django docs want you to create users. The following line goes into linking the user to the `UserPrefs` form so that a User may have additional preferences. Then we authenticate and log in.

File: vigor/forms.py (hunk 5 above)
Comment (juliancoleman) [inline vigor/forms.py]: nope :smiley: removed

File: vigor/models.py (hunk 6 above)
Comment (juliancoleman) [inline vigor/models.py]: I'll see about making an additional function to the User model to handle this a little better.

File: vigor/static/css/style.css (hunk 2 above)
Comment (juliancoleman) [inline vigor/static/css/style.css:1]: I just want to confirm that I am not experiencing this on my end.

File: vigor/ajax.py (hunk 4 above)
Comment (birkholz) [inline vigor/ajax.py:44]: Hey, sorry I haven't really had a chance to look at this.

My suggestion here is that instead of pulling data out of `req.POST`, you should be using a [ModelForm](https://docs.djangoproject.com/en/1.9/topics/forms/modelforms/). The reason to use a form is so you can validate user input before passing it to the ORM, which can be dangerous.

Here's a great SO answer relating to this
http://stackoverflow.com/questions/3523745/best-way-to-do-register-a-user-in-django

It's important to note that forms can be used even with AJAX requests. You can still take advantage of form validation, you just won't render the form.

File: vigor/models.py (hunk 6 above)
Comment (birkholz) [inline vigor/models.py]: Yea, I'd find a way to clean this up. First, separate out constants like `6.25` and `2.54` into variables, and reference the vars instead of hardcoding the same constant over and over. Then, the code can be more self-explanatory, because readers can understand what the constants are and why they're used.

I'd also split up the logic into multiple lines. Single line calculations are cool and all, but harder to read and maintain.

File: vigor/ajax.py (hunk 4 above)
Comment (juliancoleman) [inline vigor/ajax.py:44]: That's actually a really great point. I wanted to look into this, but since I was on such a short time frame to get this done, I just wanted to get this done as quickly as I could. I will render this in the next version.

File: vigor/models.py (hunk 6 above)
Comment (juliancoleman) [inline vigor/models.py]: I've pushed an update to simplify this equation, and to reduce hardcoded numbers, by assigning them to vars. The other numbers, I've left because they are a collection of other numbers precalculated. Is this okay still?

File: vigor/ajax.py (hunk 4 above)
Comment (ebramanti) [inline vigor/ajax.py:44]: @juliancoleman I say ship it for now, but add this to Waffle.

Comment (juliancoleman) [inline vigor/ajax.py:44]: Waffle it is :D

File: vigor/models.py (hunk 6 above)
Comment (birkholz) [inline vigor/models.py]: This is a lot better. 👍

**Comment by ebramanti** [comment]
@juliancoleman 👏 👏 👏 👏 great work
